cert.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/TykTechnologies/tyk/certs"
+	"github.com/TykTechnologies/tyk/config"
+
+	"github.com/gorilla/mux"
+)
+
+type APICertificateStatusMessage struct {
+	CertID  string `json:"id"`
+	Status  string `json:"status"`
+	Message string `json:"message"`
+}
+
+type APIAllCertificates struct {
+	CertIDs []string `json:"certs"`
+}
+
+// dummyGetCertificate needed because TLSConfig require setting Certificates array or GetCertificate function from start, even if it get overriden by `getTLSConfigForClient`
+func dummyGetCertificate(*tls.ClientHelloInfo) (*tls.Certificate, error) {
+	return nil, nil
+}
+
+func getTLSConfigForClient(baseConfig *tls.Config, listenPort int) func(hello *tls.ClientHelloInfo) (*tls.Config, error) {
+
+	// Supporting legacy certificate configuration
+	serverCerts := make([]tls.Certificate, 0)
+	certNameMap := make(map[string]*tls.Certificate)
+	for _, certData := range config.Global.HttpServerOptions.Certificates {
+		cert, err := tls.LoadX509KeyPair(certData.CertFile, certData.KeyFile)
+		if err != nil {
+			log.Fatalf("Server error: loadkeys: %s", err)
```

Comment:
I believe `log.Fatal` will exit the program. You either want `Error` and continue, or just `Fatal`.